interfaces/telegram.py

```diff
@@ -20,28 +21,63 @@ def _init_handlers(self):
             MessageHandler(
                 Filters.text & (Filters.entity(MessageEntity.URL) | Filters.entity(MessageEntity.TEXT_LINK)),
                 self._handle_message
+            ),
+
+            CallbackQueryHandler(
+                self._handle_mismatch_button
             )
         ]
 
         for handler in handlers:
             self.dispatcher.add_handler(handler)
 
-    def _handle_message(self, bot, update):
+    @staticmethod
+    def _handle_message(bot, update):
         text = update.message.text
         response = process_message(text)
 
+        response_keyboard = TelegramInterface.get_keyboard(update.message)
+
         if response:
             update.message.reply_markdown(
                 response,
                 quote=True,
                 disable_web_page_preview=True,
                 disable_notification=True,
+                reply_markup=response_keyboard
             )
 
+    @staticmethod
+    def _handle_mismatch_button(bot, update):
+        if TelegramInterface.ADMINS_CHAT:
```

Comment:
Not a problem, but a slight recommendation. In that cases it cleaner to do 
```
if not TelegramInterface.ADMINS_CHAT:
    return
```

That makes less indentation depth for next lines.

Reply:
ok